app/src/main/java/com/concordium/wallet/core/notifications/UpdateNotificationsSubscriptionUseCase.kt

```diff
@@ -0,0 +1,87 @@
+package com.concordium.wallet.core.notifications
+
+import android.app.Application
+import com.concordium.wallet.App
+import com.concordium.wallet.data.AccountRepository
+import com.concordium.wallet.data.backend.notifications.NotificationsBackend
+import com.concordium.wallet.data.backend.notifications.UpdateSubscriptionRequest
+import com.concordium.wallet.data.model.NotificationsTopic
+import com.concordium.wallet.data.preferences.NotificationsPreferences
+import com.concordium.wallet.data.room.Account
+import com.concordium.wallet.data.room.WalletDatabase
+import com.concordium.wallet.util.Log
+import com.google.firebase.messaging.FirebaseMessaging
+import kotlinx.coroutines.tasks.await
+
+class UpdateNotificationsSubscriptionUseCase(
+    private val accountRepository: AccountRepository,
+    private val notificationsPreferences: NotificationsPreferences,
+    private val notificationsBackend: NotificationsBackend,
+) {
+    constructor(application: Application) : this(
+        accountRepository = WalletDatabase.getDatabase(application).accountDao()
+            .let(::AccountRepository),
+        notificationsPreferences = NotificationsPreferences(application),
+        notificationsBackend = App.appCore.getNotificationsBackend(),
+    )
+
+    suspend operator fun invoke(
+        isCcdTxEnabled: Boolean? = null,
```

Comment:
You're making them nullable only to fall back to the value stored in preferences? If so, just do
```suggestion
        isCcdTxEnabled: Boolean = notificationsPreferences.areCcdTxNotificationsEnabled,
        isCis2TxEnabled: Boolean = notificationsPreferences.areCis2TxNotificationsEnabled,
```
And remove the let-run checks below